schema/messaging.proto

```diff
@@ -0,0 +1,482 @@
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\UpdateRoom
+// method.
+message UpdateRoomRequest {
+  // The room to update.
+  Room room = 1;
+
+  // The field mask to determine wich fields are to be updated. If empty, the
+  // server will assume all fields are to be updated.
+  google.protobuf.FieldMask update_mask = 2;
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\DeleteRoom
+// method.
+message DeleteRoomRequest {
+  // The resource name of the requested room.
+  string name = 1 [
+    (google.api.resource_reference) = "Room",
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\ListRooms
+// method.
+message ListRoomsRequest {
+  // The maximum number of rooms return. Server may return fewer rooms
+  // than requested. If unspecified, server will pick an appropriate default.
+  int32 page_size = 1;
+
+  // The value of google.showcase.v1alpha3.ListRoomsResponse.next_page_token
+  // returned from the previous call to
+  // `google.showcase.v1alpha3.Messaging\ListRooms` method.
+  string page_token = 2;
+}
+
+// The response message for the google.showcase.v1alpha3.Messaging\ListRooms
+// method.
+message ListRoomsResponse {
+  // The list of rooms.
+  repeated Room rooms = 1;
+
+  // A token to retrieve next page of results.
+  // Pass this value in ListRoomsRequest.page_token field in the subsequent
+  // call to `google.showcase.v1alpha3.Messaging\ListRooms` method to retrieve the
+  // next page of results.
+  string next_page_token = 2;
+}
+
+// This protocol buffer message represents a blurb sent to a chat room or
+// posted on a user profile.
+message Blurb {
+  // The resource name of the chat room.
+  string name = 1 [
+    (google.api.resource_set) = {
+      resources: [
+        {name: "RoomBlurb", path: "rooms/{room_id}/blurbs/{blurb_id}"},
+        {name: "ProfileBlurb", path: "user/{user_id}/profile/blurbs/{blurb_id}"}
+      ]
+    }];
+
+  // The resource name of the blurb's author.
+  string user = 2 [
+    (google.api.resource_reference) = "User",
+    (google.api.field_behavior) = REQUIRED];
+
+  oneof content {
+    // The textual content of this blurb.
+    string text = 3;
+
+    // The image content of this blurb.
+    bytes image = 4;
+  }
+
+  // The timestamp at which the blurb was created.
+  google.protobuf.Timestamp create_time = 5 [
+    (google.api.field_behavior) = OUTPUT_ONLY
+  ];
+
+  // The latest timestamp at which the blurb was updated.
+  google.protobuf.Timestamp update_time = 6 [
+    (google.api.field_behavior) = OUTPUT_ONLY
+  ];
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\CreateBlurb
+// method.
+message CreateBlurbRequest {
+  // The resource name of the chat room or user profile that this blurb will
+  // be tied to.
+  string parent = 1 [
+    (google.api.resource_reference) = "BlurbParent"
```

Comment:
I do not love this (spec-wise; you faithfully follow the spec here) but I have no better answer.

Reply:
Luke: I'm probably missing context; what specifically do you dislike about this?

Reply:
I agree that I don't love this, but I don't hate it. It's definitely followable.

Reply:
We only use BlurbParent in one spot, ideally it should be declared nearer or inline. But that would either add _another_ permutation to the annotation (because this approach is still needed) or restrict functionality.

It is fine as-is, if not as ideal as I might hope.

Reply:
> We only use BlurbParent in one spot. 

BlurbParent is used in two spots. IDK if that changes your feelings of this though.